packages/donation-form/src/braintree-manager/payment-providers/credit-card/credit-card.ts

```diff
@@ -2,35 +2,102 @@ import { PromisedSingleton } from '@internetarchive/promised-singleton';
     this.hostedFieldClient = options.hostedFieldClient;
     this.hostedFieldConfig = options.hostedFieldConfig;
+    this.maxRetryCount = options.maxRetryCount ?? 2;
+    this.retryInverval = (options.retryInverval ?? 1) * 1000;
+    this.loadTimeout = (options.loadTimeout ?? 6) * 1000;
+  }
+
+  private braintreeManager: BraintreeManagerInterface;
+  private hostedFieldClient: braintree.HostedFields;
+  private hostedFieldConfig: HostedFieldConfiguration;
+
+  private async createHostedFields(
+    braintreeClient: braintree.Client,
+    retryCount = 0,
+  ): Promise<braintree.HostedFields | undefined> {
+    // we mainly want to do this for retry events, but it doesn't
+    // hurt to do it on the first try
+    this.hostedFieldConfig.hostedFieldContainer.resetHostedFields();
+    try {
+      // The hosted fields have a 60 second timeout internally, but braintree
+      // support recommended setting a shorter timeout because 99% of users
+      // load the hosted fields in under 4 seconds.
+      // What we're doing here is creating a "timeout" promise
+      // and a "create hosted fields" promise and doing a `Promise.race()` to
+      // resolve when the first one finishes. If the timeout finishes first,
+      // we throw an error to trigger the retry logic. If the hosted fields
+      // finishes first, we cancel the timeout promise since we're done.
+      let timeout: number;
+      const timeoutPromise = new Promise<void>((resolve, reject) => {
+        timeout = window.setTimeout(() => {
+          reject(new Error('Timeout loading Hosted Fields'));
+        }, this.loadTimeout);
+      });
 
-    this.instance = new PromisedSingleton<braintree.HostedFields | undefined>({
-      generator: async (): Promise<braintree.HostedFields | undefined> => {
-        const braintreeClient = await this.braintreeManager.instance.get();
-        return this.hostedFieldClient.create({
+      const hostedFieldsPromise = new Promise<braintree.HostedFields | undefined>(async resolve => {
+        const fields = await this.hostedFieldClient.create({
           client: braintreeClient,
           styles: this.hostedFieldConfig.hostedFieldStyle,
           fields: this.hostedFieldConfig.hostedFieldFieldOptions,
         });
-      },
-    });
-  }
+        // clear the timeout when this finishes so we don't also get the timeout rejection
+        window.clearTimeout(timeout);
+        resolve(fields);
+      });
 
-  private braintreeManager: BraintreeManagerInterface;
-  private hostedFieldClient: braintree.HostedFields;
-  private hostedFieldConfig: HostedFieldConfiguration;
+      const result = await Promise.race([timeoutPromise, hostedFieldsPromise]);
+
+      return result as braintree.HostedFields;
+    } catch (error) {
+      if (retryCount >= this.maxRetryCount) {
+        this.emitter.emit('hostedFieldsFailed', error);
+        throw error;
+      }
+      await promisedSleep(this.retryInverval); // wait before retrying
```

Comment:
Oh this is a different timer and actually... we probably don't need this one. Originally I had a different timeout approach where I waited 1 second after the timeout to retry, but since we're already waiting 6 seconds, we don't need to wait an extra second. Going to remove.